.github/workflows/ci.yml

```
@@ -3,33 +3,35 @@ name: MolSim CI Build
 on:
   push:
     branches:
-      - dev-sheet_1
+      - dev-sheet_2
+      - dev-sheet_2-task_4
   pull_request:
     branches:
-      - dev-sheet_1
+      - master
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
-    # Step 1: Checkout the code from the repository
     - name: Checkout code
       uses: actions/checkout@v2
 
-    # Step 2: Install CMake
     - name: Install dependencies
       run: sudo apt-get update && sudo apt-get install -y cmake g++ libxerces-c-dev
 
-    # Step 3: Create build directory
     - name: Create build directory
-      run: mkdir -p build
+      run: | 
+          mkdir build
+          cd build
 
-    # Step 4: Run CMake to configure the project
-    - name: Run CMake
-      run: cmake -S . -B build
+    - name: Run CMake and Address Sanitizer
+      run: cmake -S . -B build -DCMAKE_CXX_FLAGS="-fsanitize=address -g"
 
-    # Step 5: Build the project
     - name: Build with Make
       run: cmake --build build
 
+    - name: Build and run Google Tests
```

```suggestion
    - name: Run Google Tests
```
You don't do any building!